### ISSUE TITLE
Problem to select device in blueman-sendto

### ISSUE THREAD

**Issue by ecmadrid** [blueman-project/blueman#2500]
blueman:
BlueZ:
Distribution: Arch
Desktop environment: Cinnamon

* [X ] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

Hi,

    I've localize this problem not only in Arch but in other distros too:

    1) I pair My phone with PC --> Ok.
    2) When I use blueman-sendto dialog and select a file I get devices list but if there's only one (my phone) I cannot send file. If there are more than one then I have to click in another and in my phone again to get sending file.

    In short: I need to change from another device to my device to select it.  If it's the only one then it's impossible. I thins that problem is that only select device is done on "change" event, not when "click" on it.

    Thanks a lot.

Regards,
Juan.

**Comment by infirit** [comment]
Can confirm on main. blueman-sendto relies on a signal to determine which adapter/device is selected. That signal is never sent after constructing the initial list. Not sure why this is 🤔 .

**Comment by infirit** [comment]
Found the reason, the list is populated before we connected to the signal. Quick hack to make it work below. I hope to have something nicer later.

```patch
diff --git a/blueman/gui/DeviceSelectorDialog.py b/blueman/gui/DeviceSelectorDialog.py
index 4c5ea2766..5e023bc0d 100644
--- a/blueman/gui/DeviceSelectorDialog.py
+++ b/blueman/gui/DeviceSelectorDialog.py
@@ -29,6 +29,8 @@ class DeviceSelectorDialog(Gtk.Dialog):
 
         self.selector.List.connect("device-selected", self.on_device_selected)
         self.selector.List.connect("adapter-changed", self.on_adapter_changed)
+
+        self.selector.List.populate_devices()
         if discover:
             self.selector.List.discover_devices()
 ```
